specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.2-preview/RootCauseAnalysis.json

```diff
@@ -0,0 +1,2352 @@
+        "kind": {
+          "description": "Root cause analysis kind",
+          "enum": [
+            "multidimensional"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RootCauseAnalysisType",
+            "modelAsString": true
+          }
+        },
+        "dataSetId": {
+          "format": "uuid",
+          "description": "Root cause analysis data set unique ID",
+          "type": "string",
+          "readOnly": true
+        },
+        "displayName": {
+          "description": "Root cause analysis data set name",
+          "type": "string"
+        }
+      },
+      "discriminator": "kind",
+      "example": {
+  "granularity": "daily",
+  "customInterval": 0,
+  "dimensionKeys": [
+    "Region",
+    "Version"
+  ],
+  "aggregatedIdentifier": "__SUM__",
+  "measures": [
+    {
+      "displayName": "error rate"
+    },
+    {
+      "displayName": "error count"
+    },
+    {
+      "displayName": "total call count"
+    }
+  ],
+  "displayName": "Sample Multidimensional Root Cause Analysis Dataset Name",
+  "kind": "multidimensional"
+}
+    },
+    "MultiDimRCADataSet": {
+      "required": [
+        "aggregatedIdentifier",
+        "customInterval",
+        "dimensionKeys",
+        "granularity",
+        "measures"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/RootCauseAnalysisDataSet"
+        }
+      ],
+      "properties": {
+        "granularity": {
+          "description": "Multidimensional root cause data set granularity",
+          "enum": [
+            "yearly",
+            "monthly",
+            "weekly",
+            "daily",
+            "hourly",
+            "minutely",
+            "secondly",
+            "microsecond",
+            "none"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "Granularity",
+            "modelAsString": true
+          }
+        },
+        "customInterval": {
+          "format": "int32",
+          "description": "Multidimensional root cause data set customized granularity interval. For example, if the granularity is daily and customInterval is 3, it means the data interval should be 3 days.",
+          "type": "integer"
+        },
+        "dimensionKeys": {
+          "description": "Multidimensional root cause data set dimension keys",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "aggregatedIdentifier": {
+          "description": "Identifier to describe the aggregated dimension for this data set. Only one identifier should be used.",
+          "type": "string"
+        },
+        "measures": {
+          "description": "Measures for the data",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RootCauseAnalysisMeasure"
+          }
+        }
+      },
+      "x-ms-discriminator-value": "multidimensional"
```

Comment:
Again I am not sure if I am reviewing the correct file.
"multidimensional" seems the only value here. You might consider whether this is really needed.
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#polymorphic-types

Reply:
Yes, currently we only implemented multidimensional root cause, in the future, we plan to implement cross metric root cause analysis

Reply:
Got it. When at your GA, you need to consider how backward compatibility happens to the polymorphism.

I.e., what would happen is user using e.g. v1.2 receive the new polymorphic type that only defined in newer version.

Reply:
If users are trying to use a new polymorphic type which is not in the version, our back end code has a check for the version and will throw exception for the unknown type

Reply:
It is not only about "use". User might just do a list via GET, and it might contain both known type and unknown type. Does backend include or filter out the unknown type?

Throw exception likely not an option. User should not have the API broken because someone else upload a new type.

Reply:
added a GET to list available kind

Reply:
I am not sure whether a List endpoint would notify user about the change on the possible value of kind.

Main point here is that do not take polymorphism as a fancy feature. It is not, and the feature is hard to pin-down and could cause unexpected behavior when new type added to it. Only use it when really necessary.